contracts/test/recovery-test.ts

```diff
@@ -581,4 +627,116 @@ describe("Recovery", async function () {
       fx.verificationGateway.processBundle(invalidBundle),
     ).to.be.rejectedWith("VG: Sig not verified");
   });
+
+  it("should revert when updating a bls key if signature has expired", async function () {
+    await (
+      await vg.processBundle(
+        wallet1.sign({
+          nonce: 0,
+          gas: 30_000_000,
+          actions: [],
+        }),
+      )
+    ).wait();
+
+    const invalidSignatureExpiryTimestamp =
+      (await fx.provider.getBlock("latest")).timestamp + safetyDelaySeconds;
+
+    const addressSignature = await signWalletAddress(
+      fx,
+      wallet1.address,
+      invalidSignatureExpiryTimestamp,
+      wallet2.blsWalletSigner.privateKey,
+    );
+
+    await fx.call(
+      wallet1,
+      vg,
+      "setBLSKeyForWallet",
+      [addressSignature, wallet2.PublicKey(), invalidSignatureExpiryTimestamp],
+      1,
+      30_000_000,
+    );
+
+    await fx.advanceTimeBy(safetyDelaySeconds + 1);
```

Comment:
Minor: a couple of comments around the expiry timestamp, and expectations that the first `set..` will succeed.

Reply:
Added